pulp_rpm/app/serializers/advisory.py

```diff
@@ -119,6 +119,10 @@ def create(self, validated_data):
         """
         references = validated_data.pop("references", [])
         pkglist = validated_data.pop("pkglist", [])
+
+        # detach any specified repository; attach once the advisory exists
+        repository = validated_data.pop("repository", None)
```

Comment:
Nice workaround, it didn't cross my mind when we discussed the whole problem initially.

Reply:
Yeah, I spent WAY too much time wrestling with other solution-attempts before suddenly having an "AhHA!" moment.